torch/csrc/lazy/core/tensor_impl.h

```diff
@@ -50,7 +51,7 @@ class TORCH_API LTCTensorImpl final : public c10::TensorImpl {
  private:
   void setup_size_properties();
 
-  LazyTensor tensor_;
+  LazyTensorPtr tensor_;
```

Comment:
It took me a few minutes of thinking to figure out why this is a pointer, because traditionally fields in a TensorImpl are stored inline, and turning this into an intrusive pointer introduces an indirection for seemingly no benefit.

What the benefit seems to be, is that LazyTensor may be specialized to XLATensor (in the case of XLA's backend), and now you need the indirection to be able to swap out backends.

That is all very well and fine but... does it really need to be a refcounted pointer? A `unique_ptr` to a class with a virtual destructor seems like it would be sufficient as well, since multiple references to the old LazyTensor shouldn't have been possible either.

Reply:
(psst, a comment here about why the indirection helps would be a good idea!)